extensions/rich_text_components/components.py

```diff
@@ -41,28 +52,37 @@ def validate(cls, value_dict):
         Raises:
           TypeError: if any customization arg is invalid.
         """
-        if cls.arg_schema:
-            required_attrs = cls.arg_schema
-        else:
-            required_attrs = cls.customization_args
-        required_attr_names = required_attrs.keys()
+        arg_names_to_obj_classes = {}
+        customization_arg_specs = cls.rich_text_component_specs[
+            cls.__name__]['customization_arg_specs']
+        for customization_arg_spec in customization_arg_specs:
```

Comment:
Take a look at schema_utils.normalize_against_schema(). Can/should we just use that instead?

Reply:
I think it is better to use the normalize function defined in objects.py since:

- The base object class uses `schema_utils.normalize_against_schema()` for each object and so we don't need to pass extra schema argument for each object class.
- Bool class has some preprocessing for the raw variable and it overrides the normalize method in base object class. If we use `schema_utils.normalize_against_schema()`, we have to redo this preprocessing for bool args and also if any other object tries to override normalize in base object class later, we will have to make changes here as well.

Reply:
OK, sgtm. Left one more comment.